File: decompiled/sources/freemarker/template/TransformControl.java
package freemarker.template;

import java.io.IOException;

public interface TransformControl {
    public static final int END_EVALUATION = 1;
    public static final int EVALUATE_BODY = 1;
    public static final int REPEAT_EVALUATION = 0;
    public static final int SKIP_BODY = 0;

    int afterBody() throws TemplateModelException, IOException;

    void onError(Throwable th) throws Throwable;

    int onStart() throws TemplateModelException, IOException;
}

File: decompiled/sources/freemarker/template/TransformControl.kt
package freemarker.template

import java.io.IOException

interface TransformControl {
    @Throws(TemplateModelException::class, IOException::class)
    fun afterBody(): Int

    @Throws(Throwable::class)
    fun onError(throwable: Throwable)

    @Throws(TemplateModelException::class, IOException::class)
    fun onStart(): Int

    companion object {
        const val REPEAT_EVALUATION: Int = 0
        const val SKIP_BODY: Int = 0
        const val END_EVALUATION: Int = 1
        const val EVALUATE_BODY: Int = 1
    }
}
